Remove all uses of -force-single-frontend-invocation

davezarzycki opened the pull request:
The `-force-single-frontend-invocation` flag predates WMO and is now an alias for `-whole-module-optimization`. We should use the latter and let the former fade into history.

davezarzycki commented:
@swift-ci please smoke test

harlanhaskins commented on include/swift/Option/Options.td:
```diff
@@ -935,11 +935,6 @@ def disable_batch_mode : Flag<["-"], "disable-batch-mode">,
 def wmo : Flag<["-"], "wmo">, Alias<whole_module_optimization>,
   Flags<[FrontendOption, NoInteractiveOption, HelpHidden]>;
 
-def force_single_frontend_invocation :
```
I’m fine with removing the flag but let’s leave this option here as an alias for -whole-module-optimization. We have existing clients passing -force-single-frontend-invocation and we can’t break them.

CodaFi reviewed (changes requested):
Thanks for the cleanup! I agree with Harlan's assessment: Let's leave behind an alias for now. You can even warn in the Driver about the flag being an alias. Any non-experimental flags in the driver are supposed to be considered part of a stable interface.

compnerd commented:
I think that the option alias is a requirement ... that is a driver flag.  Only the frontend flags can be removed, driver flags need to be maintained for compatibility.

compnerd commented:
@swift-ci please test Windows platform

davezarzycki committed:
Remove all uses of -force-single-frontend-invocation

The `-force-single-frontend-invocation` flag predates WMO and is now an
alias for `-whole-module-optimization`. We should use the latter and let
the former fade into history.

davezarzycki commented:
I added the old flag alias back.

@swift-ci pleas smoke test

davezarzycki commented:
@swift-ci please test Windows platform

davezarzycki commented:
@swift-ci please smoke test
